yktrace/src/swt.rs

```diff
@@ -121,36 +122,68 @@ unsafe fn yk_swt_start_tracing_impl() {
     }
 
     TRACE_BUF_CAP = TL_TRACE_INIT_CAP;
-    TRACING = true;
+    __YK_SWT_ACTIVE = true;
 }
 
 /// Record a location into the trace buffer if tracing is enabled on the current thread.
 #[do_not_trace]
 #[no_mangle]
+#[cfg(not(all(target_arch = "x86_64", target_os = "linux")))]
 unsafe extern "C" fn __yk_swt_rec_loc(symbol_name: *const i8, bb_idx: u32) {
     if !TRACING {
         return;
     }
 
+    __yk_swt_rec_loc_impl(symbol_name, bb_idx);
+}
+
+#[cfg(all(target_arch = "x86_64", target_os = "linux"))]
+global_asm!(r#"
+.globl __yk_swt_rec_loc
+.type __yk_swt_rec_loc,@function
+.section .text.__yk_swt_rec_loc,"ax",@progbits
```

Comment:
"ax" is segment flags here? executable and (what's `a`)?

Reply:
According to https://ftp.gnu.org/old-gnu/Manuals/gas-2.9.1/html_chapter/as_7.html#SEC119 it means "allocatable".

Reply:
Not much on the internet as to what that really means, but if it works, OK.

Reply:
Allocatable means that it will be mapped into memory when loading.